mops-torch/src/opsa.cpp

```diff
@@ -41,15 +41,42 @@ torch::Tensor OuterProductScatterAdd::forward(
                     torch_to_mops_1d<int32_t>(indices_output));
             });
     } else {
+#ifndef MOPS_CUDA_ENABLED
         C10_THROW_ERROR(
             ValueError,
             "outer_product_scatter_add is not implemented for device " +
                 A.device().str());
+#else
+        first_occurences = torch::zeros(
```

Comment:
this needs to be inside a `if device == CUDA`, there are other devices than CUDA & CPU =)